openquake/hazardlib/gsim/yenier_atkinson_2015.py

```diff
@@ -0,0 +1,559 @@
+                """
+                # Computing geometrical spreading, Fz.
+                h = 10 ** (-0.405 + 0.235 * rup.mag)
+                R = np.sqrt(np.amin(dists.rrup) ** 2 + (h ** 2))
+                Rref = np.sqrt(1 + (h ** 2))
+                b1 = -1.3
+                b2 = -0.5
+                Rt = 50		# Transition distance of 50km.
+                if R <= Rt:
+                        Z = R ** b1
+                else:
+                        Z = (Rt ** b1) * ((R/Rt) ** b2)
+                return np.log(Z) + ((C['b3'] + (C['b4'] * rup.mag)) * np.log(R / Rref))
+        
+        def _compute_magnitude_scaling(self, rup, C):
+                """
+                Compute magnitude-scaling term, equation (3), pag 1991.
+                """
+                # Computing Fm from equation (3), used in Fe. 
+                if rup.mag <= C['Mh']:
+                    return C['e0'] + \
+                        C['e1'] * (rup.mag - C['Mh']) + \
+                        C['e2'] * (rup.mag - C['Mh']) ** 2
+                else:
+                    return C['e0'] + \
+                        C['e1'] * (rup.mag - C['Mh'])
+
+        def _compute_stress_adjustment(self, rup, C_SA):
+                """
+                Compute stress adjustment term, equation (4), pag 1991.
+                """
+                # delta is the reference stress in bar
+                if rup.stress_drop <= 100.:
+                        e_ref_stress = C_SA['s0'] + C_SA['s1'] * rup.mag + \
+                                C_SA['s2'] * (rup.mag ** 2) + \
+                                C_SA['s3'] * (rup.mag ** 3) + \
+                                C_SA['s4'] * (rup.mag ** 4)
+                else:
+                        e_ref_stress = C_SA['s5'] + C_SA['s6'] * rup.mag + \
+                                C_SA['s7'] * (rup.mag ** 2) + \
+                                C_SA['s8'] * (rup.mag ** 3) + \
+                                C_SA['s9'] * (rup.mag ** 4)
+                        
+                return e_ref_stress * np.log(ref_stress/100)
+
+        def _get_site_amplification_linear(self, vs30, C):
+                """
+                Compute site amplification linear term,
+                equation (11), pag 1992.
+                """
+                # Analyzes each element of vs30 separately and returns an 
+                # array with the same dimensions as vs30.
+                cvc = np.ones_like(vs30)
+                for i in range(len(cvc)):
+                        cvc[i] = C['Vc']
+                
+                return_array = np.ones_like(vs30)
+
+                for i in range(len(vs30)):
+                        if vs30[i] <= cvc[i]:
+                                return_array[i] = C['c'] * np.log(vs30[i] / C['vref'])
+                        else:
+                                return_array[i] = C['c'] * np.log(cvc[i] / C['vref'])
+                return return_array
```

Comment:
As `C["Vc"]` is scalar this could be revised avoiding loops:

```python
    def _get_site_amplification_linear(self, vs30, C):
        """
        Compute site amplification linear term,
        equation (11), pag 1992.
        """
        # Inititalise with the Vs30 > Vc case as the default
        return_array = (C["c"] * np.log(C["Vc"] / C["vref"])) *\
            np.ones_like(vs30)
        # If any sites have Vs30 <= Vc then apply only to that subset
        idx = vs30 < C["Vc"]
        if np.any(idx):
            return_array[idx] = C["c"] * np.log(vs30[idx] / C["vref"])

        return return_array
```